internal/service/connect/instance_storage_config.go

```diff
@@ -0,0 +1,525 @@
+	}
+
+	tfMap, ok := tfList[0].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	result := &connect.KinesisVideoStreamConfig{
+		EncryptionConfig:     expandEncryptionConfig(tfMap["encryption_config"].([]interface{})),
+		Prefix:               aws.String(tfMap["prefix"].(string)),
+		RetentionPeriodHours: aws.Int64(int64(tfMap["retention_period_hours"].(int))),
+	}
+
+	return result
+}
+
+func exapandS3Config(tfList []interface{}) *connect.S3Config {
+	if len(tfList) == 0 || tfList[0] == nil {
+		return nil
+	}
+
+	tfMap, ok := tfList[0].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	result := &connect.S3Config{
+		BucketName:   aws.String(tfMap["bucket_name"].(string)),
+		BucketPrefix: aws.String(tfMap["bucket_prefix"].(string)),
+	}
+
+	if v, ok := tfMap["encryption_config"].([]interface{}); ok && len(v) > 0 {
+		result.EncryptionConfig = expandEncryptionConfig(v)
+	}
+
+	return result
+}
+
+func expandEncryptionConfig(tfList []interface{}) *connect.EncryptionConfig {
+	if len(tfList) == 0 || tfList[0] == nil {
+		return nil
+	}
+
+	tfMap, ok := tfList[0].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	result := &connect.EncryptionConfig{
+		EncryptionType: aws.String(tfMap["encryption_type"].(string)),
+		KeyId:          aws.String(tfMap["key_id"].(string)),
+	}
+
+	return result
+}
+
+func flattenStorageConfig(apiObject *connect.InstanceStorageConfig) []interface{} {
+	if apiObject == nil {
+		return []interface{}{}
+	}
+
+	values := map[string]interface{}{
+		"storage_type": aws.StringValue(apiObject.StorageType),
+	}
+
+	if v := apiObject.KinesisFirehoseConfig; v != nil {
+		values["kinesis_firehose_config"] = flattenKinesisFirehoseConfig(v)
+	}
+
+	if v := apiObject.KinesisStreamConfig; v != nil {
+		values["kinesis_stream_config"] = flattenKinesisStreamConfig(v)
+	}
+
+	if v := apiObject.KinesisVideoStreamConfig; v != nil {
+		values["kinesis_video_stream_config"] = flattenKinesisVideoStreamConfig(v)
+	}
+
+	if v := apiObject.S3Config; v != nil {
+		values["s3_config"] = flattenS3Config(v)
+	}
+
+	return []interface{}{values}
+}
+
+func flattenKinesisFirehoseConfig(apiObject *connect.KinesisFirehoseConfig) []interface{} {
+	if apiObject == nil {
+		return []interface{}{}
+	}
+
+	values := map[string]interface{}{
+		"firehose_arn": aws.StringValue(apiObject.FirehoseArn),
+	}
+
+	return []interface{}{values}
+}
+
+func flattenKinesisStreamConfig(apiObject *connect.KinesisStreamConfig) []interface{} {
+	if apiObject == nil {
+		return []interface{}{}
+	}
+
+	values := map[string]interface{}{
+		"stream_arn": aws.StringValue(apiObject.StreamArn),
+	}
+
+	return []interface{}{values}
+}
+
+func flattenKinesisVideoStreamConfig(apiObject *connect.KinesisVideoStreamConfig) []interface{} {
+	if apiObject == nil {
+		return []interface{}{}
+	}
+
+	// API returns <prefix>-connect-<connect_instance_alias>-contact-
+	prefixRaw := aws.StringValue(apiObject.Prefix)
+	regexPatten := regexp.MustCompile(`-connect-.*-contact-`)
```

Comment:
```suggestion
	regexPattern := regexp.MustCompile(`-connect-.*-contact-`)
```

Reply:
Resolved in 5da39fd7eb09a288c5f3a3509575fe94ad66a36e